#!/usr/bin/env python
PACKAGE = "franka_example_controllers"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
#          name                   paramtype level desciption                    default=? min  max
gen.add("JointImpedanceController", str_t, 0, "JointImpedanceController", "JointImpedanceController")

gen.add("Joint_r1", double_t, 0, "Joint_r1", 10, 0, 1000)#
gen.add("Joint_r2", double_t, 0, "Joint_r2", 10, 0, 1000)#
gen.add("Joint_r3", double_t, 0, "Joint_r3", 10, 0, 1000)#
gen.add("Joint_r4", double_t, 0, "Joint_r4", 10, 0, 1000)#
gen.add("Joint_rd", double_t, 0, "Joint_rd", 10, 0, 1000)#
gen.add("Joint_Md", double_t, 0, "Joint_Md", 1, 0, 100)#
gen.add("Joint_Dd", double_t, 0, "Joint_Dd", 2,  0, 600)#
gen.add("Joint_Kd", double_t, 0, "Joint_Kd", 5,  0, 600)#

gen.add("CartesianImpedanceController", str_t, 0, "CartesianImpedanceController", "CartesianImpedanceController")

gen.add("Cartesian_r", double_t, 0, "Cartesian_r", 30, 0, 100)#
gen.add("Cartesian_Md", double_t, 0, "Cartesian_Md", 1, 0, 10)#
gen.add("Cartesian_Dd", double_t, 0, "Cartesian_Dd", 1,  0, 100)#
gen.add("Cartesian_Kd", double_t, 0, "Cartesian_Kd", 2,  0, 200)#

#pkgname nodename name
exit(gen.generate(PACKAGE, "dynamic_compliance", "impedance_controller_param"))
